HealthVault/Classes/Models/MHVDateTimeBase.m

```diff
@@ -0,0 +1,51 @@
+//
+// MHVDateTimeBase.m
+// MHVLib
+//
+// Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import "MHVDateTimeBase.h"
+
+@implementation MHVDateTimeBase
+
+- (instancetype)initWithObject:(id)object objectParameters:(NSObject *)parameters
+{
+    NSDateFormatter *dateFormatter = [NSDateFormatter new];
+    [dateFormatter setDateFormat:[self getDateFormatString]];
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+    
+    self = [super init];
+    if (self)
+    {
+        self.date = [dateFormatter dateFromString:object];
```

Comment:
Not a limitation of the auto-gen code. The function is inherited from MHVModelBase - I should actually remove the declaration from the .h file. Since it's overriding, I probably shouldn't change the parameter type. I'll add a check for the class.